tests/drivers/clock_control/adsp_clock/src/main.c

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/ztest.h>
+#include <zephyr/drivers/clock_control/clock_control_adsp.h>
+#include <zephyr/drivers/clock_control.h>
+
+static void check_clocks(struct adsp_clock_info *clocks, uint32_t freq_idx)
+{
+	int i;
+	unsigned int num_cpus = arch_num_cpus();
+
+	for (i = 0; i < num_cpus; i++) {
+		zassert_equal(clocks[i].current_freq, freq_idx, "");
```

Comment:
Maybe add some descriptive message to assert?

Reply:
zassert_equal & zassert_not_null already have messages, but however, if you have any ideas or propositions, please let me know, and I will update it :)